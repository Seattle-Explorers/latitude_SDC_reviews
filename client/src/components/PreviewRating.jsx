import React from 'react';

import {packetCategory} from './Rating';

// Styling imports
import { RatingPreviewContainer, CategoryContainer, RatingGrid } from './styles/staticStyles.style'
import { TextContainer, RatingText } from './styles/generalUse.style';
import { Bar } from './styles/propStyles.style';

const PreviewRating = ({ overview }) => {
  const { cleanAvg, commAvg, accuracyAvg, valueAvg, locationAvg, checkinAvg } = overview;

  const rating = [cleanAvg, accuracyAvg, commAvg, locationAvg, checkinAvg, valueAvg];

  const categoryData = packetCategory(rating);

  return (
    <RatingPreviewContainer>
      {categoryData.map((category) => (
        <CategoryContainer key={`preview ${category.name}`}>
          <TextContainer>{category.name}</TextContainer>
          <RatingGrid>
            <Bar margin='5px'>
              <Bar width={`${category.percent}%`} color='black' height='100%'/>
            </Bar>
          <RatingText>{category.rating}</RatingText>
          </RatingGrid>
        </CategoryContainer>
      ))}
    </RatingPreviewContainer>
  )
}

export default PreviewRating;
