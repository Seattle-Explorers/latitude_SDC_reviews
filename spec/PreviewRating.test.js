import 'jsdom-global/register';
import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-enzyme';
import 'jest-styled-components';
import { mount } from 'enzyme';
import styled from 'styled-components';

import { cleanAvg, commAvg, accuracyAvg, valueAvg, locationAvg, checkinAvg } from '../__mocks__/mockData.js';
import PreviewRating from '../client/src/components/PreviewRating.jsx';

describe('Preview Rating Components', () => {

  const stats = {
    cleanAvg: cleanAvg,
    commAge: commAvg,
    accuracyAvg: accuracyAvg,
    valueAvg: valueAvg,
    locationAvg: locationAvg,
    checkinAvg: checkinAvg
  }

  let wrapper;

  beforeEach(() => {
    wrapper = mount(<PreviewRating overview={stats} />)
  })

  it('should render', () => {
    const component = <PreviewRating overview={stats} />;

    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  })
})