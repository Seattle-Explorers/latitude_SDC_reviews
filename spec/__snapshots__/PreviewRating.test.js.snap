// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Preview Rating Components should render 1`] = `
.c0 {
  display: grid;
  grid-template-columns: repeat(2,auto);
  margin-right: 50px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin-left: 50px;
  margin-right: 50px;
  margin-top: 15px;
  width: 100%;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  margin-right: 10px;
  width: 75%;
}

.c2 {
  width: 25%;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c6 {
  font-size: 13px;
  font-weight: bolder;
}

.c4 {
  margin-top: 5px;
  width: 33%;
  height: 0.25em;
  float: left;
  border-radius: 15%;
  background-color: #B3B3B3;
  margin-right: 5px;
}

.c5 {
  margin-top: 0;
  width: 20%;
  height: 100%;
  float: left;
  border-radius: 15%;
  background-color: black;
  margin-right: 5px;
}

.c7 {
  margin-top: 0;
  width: 24%;
  height: 100%;
  float: left;
  border-radius: 15%;
  background-color: black;
  margin-right: 5px;
}

.c8 {
  margin-top: 0;
  width: 52%;
  height: 100%;
  float: left;
  border-radius: 15%;
  background-color: black;
  margin-right: 5px;
}

.c9 {
  margin-top: 0;
  width: 80%;
  height: 100%;
  float: left;
  border-radius: 15%;
  background-color: black;
  margin-right: 5px;
}

.c10 {
  margin-top: 0;
  width: 38%;
  height: 100%;
  float: left;
  border-radius: 15%;
  background-color: black;
  margin-right: 5px;
}

.c11 {
  margin-top: 0;
  width: 32%;
  height: 100%;
  float: left;
  border-radius: 15%;
  background-color: black;
  margin-right: 5px;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      Cleaniness
    </div>
    <div
      className="c3"
    >
      <div
        className="c4"
      >
        <div
          className="c5"
          color="black"
          height="100%"
          width="20%"
        />
      </div>
      <span
        className="c6"
      >
        1
      </span>
    </div>
  </div>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      Accuracy
    </div>
    <div
      className="c3"
    >
      <div
        className="c4"
      >
        <div
          className="c7"
          color="black"
          height="100%"
          width="24%"
        />
      </div>
      <span
        className="c6"
      >
        1.2
      </span>
    </div>
  </div>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      Communication
    </div>
    <div
      className="c3"
    >
      <div
        className="c4"
      >
        <div
          className="c8"
          color="black"
          height="100%"
          width="52%"
        />
      </div>
      <span
        className="c6"
      >
        2.6
      </span>
    </div>
  </div>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      Location
    </div>
    <div
      className="c3"
    >
      <div
        className="c4"
      >
        <div
          className="c9"
          color="black"
          height="100%"
          width="80%"
        />
      </div>
      <span
        className="c6"
      >
        4
      </span>
    </div>
  </div>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      Check-in
    </div>
    <div
      className="c3"
    >
      <div
        className="c4"
      >
        <div
          className="c10"
          color="black"
          height="100%"
          width="38%"
        />
      </div>
      <span
        className="c6"
      >
        1.9
      </span>
    </div>
  </div>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      Value
    </div>
    <div
      className="c3"
    >
      <div
        className="c4"
      >
        <div
          className="c11"
          color="black"
          height="100%"
          width="32%"
        />
      </div>
      <span
        className="c6"
      >
        1.6
      </span>
    </div>
  </div>
</div>
`;
